### PR TITLE
Add support for Traffic switching

### PR DESCRIPTION
**Pull request by mikkeloscar**
Note: Rebased on top of #57 

This adds support for traffic switching (ingress) between deployments through the new command:

```
zdeploy traffic <application> <release> <percent>
```

It also enhances the `apply` command to look for previous release of a service and provide the `prev_release` variable which can be used in ingress manifests like this to ensure that the a new release of an application want get traffic right away:

```yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: "{{ application }}"
  labels:
    application: "{{ application }}"
  annotations:
    zalando.org/backend-weights: |
      {"{{ application }}-{{ prev_release }}": 100}
spec:
  rules:
  - host: "{{ application }}.playground.zalan.do"
    http:
      paths:
      - backend:
          serviceName: "{{ application }}-{{ release }}"
          servicePort: 80
      - backend:
          serviceName: "{{ application }}-{{ prev_release }}"
          servicePort: 80
```

Initially we wanted to use `{{ application }}-{{ version }}-{{ release }}` for service names, but this doesn't work if `version` includes dots as the service name must the a valid DNS name.

```
a DNS-1035 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?')
```

That's why I only included `release` in the service name. Alternatively we could transform the `version` to something that is valid, but I don't know if this makes sense?

### CODE REVIEW

File: zalando_deploy_cli/cli.py [hunk 1]
```diff
@@ -479,6 +506,65 @@ def apply_autoscaling(config, template, application, version, release, parameter
...
 
 
+@cli.command('traffic')
+@application_argument
+@release_argument
+@click.argument('percent', type=int)
+@click.pass_obj
+@click.option('--execute', is_flag=True)
+def traffic(config, application, release, percent, execute):
+    if percent < 0 or percent > 100:
```
Comment (hjacobs) [inline zalando_deploy_cli/cli.py]: much nicer: `click.IntRange` (http://click.pocoo.org/5/options/#range-options)

Comment (mikkeloscar) [inline zalando_deploy_cli/cli.py]: Thanks, fixed.

**Comment by hjacobs** [comment]
@mikkeloscar please rebase on master..

**Comment by hjacobs** [comment]
:+1:

**Comment by tuxlife** [comment]
:+1:

**Pull request merged**